### PR TITLE
bpo-32533: Fixed thread-safety of error handling in _ssl.

### PR DESCRIPTION
**Pull request by zooba**
https://bugs.python.org/issue32533

### CODE REVIEW

File: Modules/_ssl.c [hunk 1]
```diff
@@ -455,20 +459,19 @@ static PyTypeObject PySSLSocket_Type;
 static PyTypeObject PySSLMemoryBIO_Type;
 static PyTypeObject PySSLSession_Type;
 
+static inline _PySSLError _PySSL_errno(int failed, const SSL *ssl, int retcode)
+{
+    _PySSLError err = { 0 };
```
Comment (tiran) [inline Modules/_ssl.c:464]: IIRC that's not safe in C code. You are allocating a struct on the stack and then return the stack allocated struct to the caller. Once you return to the caller, the stack allocated memory becomes invalid and may be reused in another function call.

See https://cwe.mitre.org/data/definitions/562.html

**Review by tiran — DISMISSED** [review]

File: Modules/_ssl.c (hunk 1 above)
Comment (zooba) [reply Modules/_ssl.c:464]: It returns a copy, not a pointer.

**Comment by zooba** [comment]
I have made the requested changes; please review again

**Comment by ned-deily** [comment]
@tiran, will you time to review the changes you requested?

**Pull request merged**